File: frontend/src/types/brief.ts
// src/types/brief.ts
// Types for the Risk Brief engine (Phase 10).
// Mirrors app/schemas/brief.py BriefPayload shape.

export interface SignalRow {
  type: string;
  exposure_delta?: number | null;
  exposure_driver?: string | null;
  failed_controls?: number | null;
  incident_count?: number | null;
  incident_area?: string | null;
}

export interface BriefTableRow {
  id: string;
  description: string;
  level?: string | null;
  delta?: number | null;
  days_logged?: number | null;
  days_since_review?: number | null;
  action?: string | null;
}

export interface BriefTables {
  volatile: BriefTableRow[];
  high_critical: BriefTableRow[];
  stale: BriefTableRow[];
}

export interface OutreachItem {
  owner_name: string;
  risk_id?: string | null;
  reason: string;
  message: string;
  mailto?: string | null;
}

export interface DailyException {
  is_empty: boolean;
  empty_message?: string | null;
  item_count: number;
}

export interface WeeklyDigest {
  health_wow: number;
  health_now: number;
  health_prev: number;
  top_movers_count: number;
  overdue_reviews_count: number;
  overdue_actions_count: number;
}

export interface BriefMeta {
  cadence_sections: string[];
  workspace_name: string;
  is_warming_up: boolean;
  generated_at: string;
}

export interface BriefPayload {
  ok: boolean;
  skip: boolean;
  reason?: string | null;
  reader?: { first_name: string; email: string } | null;
  meta?: BriefMeta | null;
  greeting?: string | null;
  residual_change_summary?: string | null;
  critical_count: number;
  signal_rows: SignalRow[];
  recommended_action?: string | null;
  tables?: BriefTables | null;
  outreach: OutreachItem[];
  daily_exception?: DailyException | null;
  weekly?: WeeklyDigest | null;
}

export interface SendTestBriefRequest {
  to_email: string;
}

export interface SendTestBriefResponse {
  ok: boolean;
  reason?: string | null;
}

File: backend/app/schemas/brief.py
# app/schemas/brief.py
# Pydantic schemas for the Risk Brief engine (Phase 10).
# Derived from BriefService.gs api_buildBriefPayload() return structure.

from pydantic import BaseModel


class SignalRow(BaseModel):
    type: str
    exposure_delta: float | None = None
    exposure_driver: str | None = None
    failed_controls: int | None = None
    incident_count: int | None = None
    incident_area: str | None = None


class BriefTableRow(BaseModel):
    id: str
    description: str
    level: str | None = None
    delta: float | None = None
    days_logged: int | None = None
    days_since_review: int | None = None
    action: str | None = None


class BriefTables(BaseModel):
    volatile: list[BriefTableRow] = []
    high_critical: list[BriefTableRow] = []
    stale: list[BriefTableRow] = []


class OutreachItem(BaseModel):
    owner_name: str
    risk_id: str | None = None
    reason: str
    message: str
    mailto: str | None = None


class DailyException(BaseModel):
    is_empty: bool
    empty_message: str | None = None
    item_count: int = 0


class WeeklyDigest(BaseModel):
    health_wow: float = 0.0
    health_now: float = 0.0
    health_prev: float = 0.0
    top_movers_count: int = 0
    overdue_reviews_count: int = 0
    overdue_actions_count: int = 0


class BriefReader(BaseModel):
    first_name: str
    email: str


class BriefMeta(BaseModel):
    cadence_sections: list[str]
    workspace_name: str
    is_warming_up: bool
    generated_at: str


class BriefPayload(BaseModel):
    ok: bool
    skip: bool = False
    reason: str | None = None
    reader: BriefReader | None = None
    meta: BriefMeta | None = None
    greeting: str | None = None
    residual_change_summary: str | None = None
    critical_count: int = 0
    signal_rows: list[SignalRow] = []
    recommended_action: str | None = None
    tables: BriefTables | None = None
    outreach: list[OutreachItem] = []
    daily_exception: DailyException | None = None
    weekly: WeeklyDigest | None = None


class SendTestBriefRequest(BaseModel):
    to_email: str